compiler/rustc_middle/src/ty/codec.rs

```diff
@@ -39,7 +39,7 @@ impl<'tcx, E: TyEncoder<'tcx>> EncodableWithShorthand<'tcx, E> for Ty<'tcx> {
 
     #[inline]
     fn variant(&self) -> &Self::Variant {
-        self.kind()
+        &self.kind
```

Comment:
hmm, can you add a fixme here, I would like to potentially solve this differently.

Might just use
```suggestion
        // FIXME(@lcnr): Look into instead returning `Self::Variant` directly here
        &self.kind
```